File: Iniciante/1117/1117.c
#include <stdio.h>
 
int main() {
 
    double nota1, nota2, media;
    
    do {
        scanf("%lf", &nota1);
        if (nota1 < 0 || nota1 > 10)
            printf("nota invalida\n");
    } while (nota1 < 0 || nota1 > 10);
    
    do {
        scanf("%lf", &nota2);
        if (nota2 < 0 || nota2 > 10)
            printf("nota invalida\n");
    } while (nota2 < 0 || nota2 > 10);
    
    media = (nota1 + nota2)/2;
    printf("media = %0.2lf\n", media);
    
    return 0;
}

File: Python/Iniciante/1117.py
# Problema 1117 - Beecrowd - Iniciante - Nível 1
# Laço que recebe nota1 na entrada e para apenas se o valor estiver no intervalo [0, 10]
while (nota1 := float(input())) < 0 or nota1 > 10: print('nota invalida')
# Laço que recebe nota2 na entrada e para apenas se o valor estiver no intervalo [0, 10]
while (nota2 := float(input())) < 0 or nota2 > 10: print('nota invalida')
# Mostrando resultado da média na tela
print(f'media = {(nota1+nota2)/2.0}')
